builder/openstack/run_config.go

```diff
@@ -247,10 +252,15 @@ func (c *RunConfig) Prepare(ctx *interpolate.Context) []error {
 		}
 	}
 
-	if c.SourceImage == "" && c.SourceImageName == "" && c.SourceImageFilters.Filters.Empty() {
-		errs = append(errs, errors.New("Either a source_image, a source_image_name, or source_image_filter must be specified"))
-	} else if len(c.SourceImage) > 0 && len(c.SourceImageName) > 0 {
-		errs = append(errs, errors.New("Only a source_image or a source_image_name can be specified, not both."))
+	if c.SourceImage == "" && c.SourceImageName == "" && c.ExternalSourceImageURL == "" && c.SourceImageFilters.Filters.Empty() {
+		errs = append(errs, errors.New("Either a source_image, a source_image_name, an external_source_image_url or source_image_filter must be specified"))
+	} else if (len(c.SourceImage) > 0 && len(c.SourceImageName) > 0) || (len(c.SourceImage) > 0 && len(c.ExternalSourceImageURL) > 0) || (len(c.SourceImageName) > 0 && len(c.ExternalSourceImageURL) > 0) {
+		errs = append(errs, errors.New("Only a source_image, a source_image_name or an external_source_image_url can be specified, not multiple."))
+	}
```

Comment:
I had to read this if condition a couple of times to understand, because of the length check after content check.
Do you think is better this way? 👇 
```suggestion
	} else {
		if c.SourceImage != "" {
			if c.SourceImageName != "" || c.ExternalSourceImageURL != "" {
				errs = append(errs, errors.New("Only a source_image, a source_image_name or an external_source_image_url can be specified, not multiple."))
			}
		}
		
		if c.SourceImageName != "" {
			if c.ExternalSourceImageURL != "" {
				errs = append(errs, errors.New("Only a source_image_name or an external_source_image_url can be specified, not multiple."))
			}
		}
	}
```

Reply:
yes, this is a bit of a messy section. You could also introduce some temporary variables like `hasOnlySourceImage`, `hasOnlySourceImageName`, etc. and then use those temp variables in the if

Reply:
take a look at the new version. I think it's more readable

Reply:
looks good! Thanks